# 03 01 Exercícios Pandas
---

In [1]:
# faça seus imports aqui 
import pandas as pd

## EX01

Criação e métodos de séries.

In [2]:
temperaturas = [15.4, 18.0, 22.5, 20.0, 20.3, 25.0]

__A.__ Crie uma série com os dados da lista fornecida. Utilize pd.Series e atribua a serie à variável s_temperaturas

__B.__ Imprima o tipo do objeto s_temperaturas

pandas.core.series.Series

__C.__ Imprima o objeto s_temperaturas

0    15.4
1    18.0
2    22.5
3    20.0
4    20.3
5    25.0
dtype: float64

__D.__ Imprima o numéro de linhas do objeto

(6,)

__E.__ imprima as 2 primeiras linhas

0    15.4
1    18.0
dtype: float64

__F__. imprima as 2 últimas linhas

4    20.3
5    25.0
dtype: float64

__G__. imprima o maior valor

25.0

__H__. imprima o menor valor

15.4

__J__. utilize o método describe para visualizar estatísticas básicas da série

count     6.0000
mean     20.2000
std       3.3544
min      15.4000
25%      18.5000
50%      20.1500
75%      21.9500
max      25.0000
dtype: float64

## EX02

Filtros e expressões em séries.

__A.__ Crie uma série com as variáveis temperatura e indice definidas abaixo

In [14]:
temperaturas = [15.4, 18.0, 22.5, 20.0, 20.3, 25.0]
indice = pd.to_datetime(['2017-05-03','2017-05-04','2017-05-04','2017-05-05','2017-05-06','2017-05-07'])

s_temperaturas = 

2017-05-03    15.4
2017-05-04    18.0
2017-05-04    22.5
2017-05-05    20.0
2017-05-06    20.3
2017-05-07    25.0
dtype: float64

__B.__ Filtre os registros pelo índice 2017-05-04

2017-05-04    18.0
2017-05-04    22.5
dtype: float64

__C.__ Filtre os registros pelo intervalo 2017-05-04 a 2017-05-06

2017-05-04    18.0
2017-05-04    22.5
2017-05-05    20.0
2017-05-06    20.3
dtype: float64

__D.__ Filtre os registros das posições 1 a 3

2017-05-03    15.4
2017-05-04    18.0
2017-05-04    22.5
dtype: float64

__E.__ Filtre os registros com temperatura maior que 20

2017-05-04    22.5
2017-05-06    20.3
2017-05-07    25.0
dtype: float64

## EX03

Criação e características de DataFrames.

__A.__ Crie um dataframe utilizando a lista de tuplas abaixo.

In [19]:
# utilize esta variável para o nome das colunas
nomes_colunas = ['Espécie', 'Idade', 'Visitas', 'Prioridade']
clinica_veterinaria = [
    ('Gato', 6.0, 12, 'Sim'),
    ('Gato', 3.0, 23, 'Sim'),
    ('Cachorro', 10.0, 14, 'Sim'),
    ('Ramster', 4.0, 5, 'Não'),
    ('Tartaruga', 10.0, 11, 'Sim'),
    ('Tartaruga', 3.0, 2, 'Sim'),
    ('Gato', 12.0, 3, 'Não'),
    ('Ramster', 11.0, 26, 'Não'),
    ('Gato', 5.5, 29, 'Não'),
    ('Cachorro', 2.0, 28, 'Sim'),
    ('Coelho', 5.5, 1, 'Sim'),
    ('Tartaruga', 4.0, 21, 'Não'),
    ('Ramster', 1.0, 28, 'Sim'),
    ('Tartaruga', 3.5, 18, 'Não'),
    ('Ramster', 3.0, 20, 'Não'),
    ('Gato', 2.0, 16, 'Não'),
    ('Tartaruga', 10.5, 29, 'Não'),
    ('Cachorro', 10.5, 26, 'Sim'),
    ('Coelho', 10.0, 26, 'Sim'),
    ('Ramster', 9.5, 2, 'Não')
]

# crie o dataframe e atribua à variável 'df_clinica_veterinaria'


# exiba as primeiras 5 linhas


Espécie  Idade  Visitas Prioridade
0       Gato    6.0       12        Sim
1       Gato    3.0       23        Sim
2   Cachorro   10.0       14        Sim
3    Ramster    4.0        5        Não
4  Tartaruga   10.0       11        Sim

__B.__ Crie uma nova coluna com o resultado do número de visitas dividido pela idade. O nome da colunas deve ser "Visitas/Idade"

Espécie  Idade  Visitas Prioridade  Visitas/Idade
0       Gato    6.0       12        Sim       2.000000
1       Gato    3.0       23        Sim       7.666667
2   Cachorro   10.0       14        Sim       1.400000
3    Ramster    4.0        5        Não       1.250000
4  Tartaruga   10.0       11        Sim       1.100000

__C.__ Ordene o Dataframe pela nova coluna "Visitas/Idade" em ordem decrescente

Espécie  Idade  Visitas Prioridade  Visitas/Idade
12   Ramster    1.0       28        Sim      28.000000
9   Cachorro    2.0       28        Sim      14.000000
15      Gato    2.0       16        Não       8.000000
1       Gato    3.0       23        Sim       7.666667
14   Ramster    3.0       20        Não       6.666667

__D.__ Selecione todos os registros com cachorros

Espécie  Idade  Visitas Prioridade  Visitas/Idade
2   Cachorro   10.0       14        Sim        1.40000
9   Cachorro    2.0       28        Sim       14.00000
17  Cachorro   10.5       26        Sim        2.47619

__E.__ Selecione todos os registros com gatos com idade acima de 4 anos

Espécie  Idade  Visitas Prioridade  Visitas/Idade
0    Gato    6.0       12        Sim       2.000000
6    Gato   12.0        3        Não       0.250000
8    Gato    5.5       29        Não       5.272727

__F.__ Qual o total de visitas registradas na clínica?

340

__G.__ Existe algum Ramster com prioridade?

Espécie  Idade  Visitas Prioridade  Visitas/Idade
12  Ramster    1.0       28        Sim           28.0

__H.__ Qual a maior idade por espécie?

Espécie
Cachorro     10.5
Coelho       10.0
Gato         12.0
Ramster      11.0
Tartaruga    10.5
Name: Idade, dtype: float64

__I.__ Qual a média de idade por espécie?

Espécie
Cachorro     7.50
Coelho       7.75
Gato         5.70
Ramster      5.70
Tartaruga    6.20
Name: Idade, dtype: float64

In [30]:
# Gerador de clientes para clínica veterinária

sorteio_animais = ['Gato', 'Cachorro', 'Coelho', 'Ramster', 'Tartaruga']
sorteio_idades = list(np.arange(0.5, 13.0, step=0.5))
sorteio_visitas = list(np.arange(1, 30))
sorteio_prioridade = ['Sim', 'Não']

clinica_veterinaria = [(np.random.choice(sorteio_animais),
                        np.random.choice(sorteio_idades),
                        np.random.choice(sorteio_visitas),
                        np.random.choice(sorteio_prioridade),
                        ) for i in range(20)
                        ]
clinica_veterinaria

[('Coelho', 8.5, 29, 'Sim'),
 ('Tartaruga', 6.5, 26, 'Não'),
 ('Coelho', 8.5, 25, 'Não'),
 ('Coelho', 3.0, 21, 'Sim'),
 ('Gato', 1.0, 24, 'Não'),
 ('Coelho', 9.5, 6, 'Não'),
 ('Tartaruga', 7.5, 29, 'Sim'),
 ('Coelho', 5.0, 2, 'Sim'),
 ('Tartaruga', 10.5, 2, 'Não'),
 ('Tartaruga', 9.5, 18, 'Sim'),
 ('Tartaruga', 10.5, 16, 'Sim'),
 ('Tartaruga', 6.5, 11, 'Sim'),
 ('Coelho', 9.0, 2, 'Não'),
 ('Coelho', 5.0, 27, 'Não'),
 ('Gato', 12.5, 27, 'Não'),
 ('Gato', 11.5, 6, 'Não'),
 ('Tartaruga', 7.5, 25, 'Não'),
 ('Gato', 3.0, 14, 'Não'),
 ('Gato', 3.5, 17, 'Não'),
 ('Tartaruga', 3.0, 25, 'Não')]

___
__Material produzido para o curso__:
* Introdução à Análise de Dados com Python

__Autor__:
* Fernando Sola Pereira

__Revisão__:
* 1.1